### ISSUE TITLE
Rethinking Theming support

### ISSUE THREAD

**Issue by emmenko** [commercetools/ui-kit#1661]
We should also probably rethink a bit how we plan and want to continue support theming.
At the moment is more of a "hidden" feature but we do rely on it in other places.
It's also not super consistent as only a bunch of components support that.

Apparently, in some recent emotion updates there have been some "regressions" (#1660 and https://github.com/commercetools/commercetools-docs-kit/pull/745) which forces us to fix that here.

I think we need to make a decision how we want to continue the theming support.

PS: have a read at https://epicreact.dev/css-variables/ as an inspiration for an alternative approach (assuming it would be possible in our setup).

**Comment by bradd123** [comment]
Has any discussion happened regarding this?

**Comment by tdeekens** [comment]
Nope. It hasn't been a priority and we do not need it ourselves at the moment.
